src/borg/archive.py

```diff
@@ -1034,28 +1034,35 @@ def identify_key(self, repository):
         return key_factory(repository, cdata)
 
     def verify_data(self):
+        def chunker(it, size):
+            while True:
+                result = []
+                try:
+                    for i in range(size):
+                        result.append(next(it))
+                except StopIteration:
+                    yield result
+                    raise StopIteration
+                else:
+                    yield result
```

Comment:
that sounds like you passed a view instead of a iterator

wrap an call to iter around any iterator to be sure
an iterator returns self, a view will return a view iterator

once you have the view iterator, it will not reset